src/View/Widget/MultiCheckboxWidget.php

```diff
@@ -113,10 +115,26 @@ public function render(array $data, ContextInterface $context)
             'idPrefix' => null,
             'templateVars' => []
         ];
-        $out = [];
         $this->_idPrefix = $data['idPrefix'];
         $this->_clearIds();
+        return implode('', $this->_renderInputs($data, $context));
+    }
+
+    protected function _renderInputs($data, $context)
+    {
+        $out = [];
         foreach ($data['options'] as $key => $val) {
+            // Grouped inputs in a fieldset.
+            if (is_string($key) && is_array($val) && !isset($val['text'], $val['value'])) {
+                $inputs = $this->_renderInputs(['options' => $val] + $data, $context);
+                $legend = $this->_templates->format('legend', ['text' => $key]);
+                $out[] = $this->_templates->format('fieldset', [
```

Comment:
If this input widget was used with input() it would get a containing element from that method. When used in isolation, the developer could wrap the widget with raw HTML.

Are you suggesting adding a template like `multicheckboxcontainer` and `multicheckboxtitle` that default to a fieldset/legend?